File: Project/Data/Models/Shift.cs
using Project.Data.Enums;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Project.Data.Models
{
    public class Shift
    {
        [Key]
        public int Id { get; set; }

        [Required]
        public string UserId { get; set; }

        [ForeignKey(nameof(UserId))]
        public User User { get; set; }

        [Required]
        public ShiftType ShiftType { get; set; }

        public DateTime? StartTime { get; set; }
        public DateTime? EndTime { get; set; }
        public bool IsActive { get; set; } = true;
        public string? Notes { get; set; }
        public string CreatedBy { get; set; }
        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    }
}

File: frontend/src/types/shift.ts
export type ShiftType = 'Morning' | 'Evening' | 'Night' | 1 | 2 | 3;

export interface ShiftDto {
  id: number;
  userId: string;
  userName?: string;
  shiftType: ShiftType | number;
  startTime?: string;
  endTime?: string;
  isActive: boolean;
  notes?: string;
  createdBy?: string;
  createdAt?: string;
  updatedBy?: string;
  updatedAt?: string;
  totalOrders?: number;
  totalAmount?: number;
}

export interface CreateShiftDto {
  userId: string;
  shiftType: ShiftType;
  notes?: string;
}


